clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2636,6 +2616,72 @@ In OpenCL C, this attribute is available in GNU spelling
 
   __kernel __attribute__((reqd_work_group_size(8, 16, 32))) void test() {}
 
+The order of the arguments to reqd_work_group_size in SYCL and OpenCL are
```

Comment:
I've read this a few times and I keep getting lost in the x's and y's and preserved definitions, etc. I think we need this to be super clear documentation, so I think we should try to rewrite it in terms of index increment speed. e.g.,
```
The the arguments to reqd_work_group_size are ordered based on which index increments the fastest. In OpenCL, the first argument is the index that increments the fastest, and in SYCL, the last argument is the index that increments the fastest. 

In OpenCL, all three arguments are required.

In SYCL, the attribute accepts either one, two, or three arguments; in each form, the last (or only) argument is the index that increments fastest. The number of arguments passed to the attribute must match the dimensionality of the kernel the attribute is applied to.

If the reqd_work_group_size attribute is specified on a declaration along with num_simd_work_items, the required work group size specified by num_simd_work_items must evenly divide the index that increments fastest in the reqd_work_group_size attribute. 

<tons of examples>
```
WDYT?

Reply:
@AaronBallman, sorry about the long description. I will update the documentation. Thanks.